mesonbuild/interpreterbase/interpreterbase.py

```diff
@@ -127,7 +127,10 @@ def sanity_check_ast(self) -> None:
             found = p
             for parent in p.parents:
                 if (parent / 'meson.build').is_file():
-                    found = parent
+                    with open(parent / 'meson.build', 'r') as f:
```

Comment:
Do we need to add `encoding='utf-8'` here?

Reply:
Yeah, I noticed the custom_lint error from my fork...